eiffel-syntax-and-usage/glossary.md

```diff
@@ -0,0 +1,102 @@
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+--->
+
+# Eiffel Glossary
+This a non-exhaustive, alphabetical list of terms used in the [Eiffel vocabulary](../eiffel-vocabulary/) and its documentation.
+
+ - [Terms Used in Event Definitions](#terms-used-in-event-definitions)
+   - [Activity](#activity)
+   - [Artifact](#artifact)
+   - [Composition](#composition)
+   - [Confidence Level](#confidence-level)
+   - [Environment](#environment)
+   - [Event](#event)
+   - [Source Change](#source-change)
+   - [Submit](#submit)
+
+ - [Terms Used in Documentation](#terms-used-in-documentation)
+   - [Pipeline](#pipeline)
+
+## Terms Used in Event Definitions
+These terms are either part of the event names in the Eiffel Vocabulary, or they are part of parameter names or values in the events. Users of Eiffel should be able to rely on these terms to be kept, unless a major change to event types are done.
+
+### Activity
+An _activity_ is any kind of action in a CI/CD system, normally triggered by an operation done in an SCM system or by a previous activity. The Eiffel protocol declares events related to activities to describe what is going on, what has finished, when did it happen and with what outcome.
+
+Activities are hierarchical. Activities could be whole [pipelines](#pipeline), or sequential or parallel parts thereof. Such pipeline parts could for example be build activities or test activities.
+
+An activity could also be an SCM operation such as a manual code review or automated source change check.
+
+Some activity types have their own specific events, such as EiffelTestCase\*Events notifying test case executions, while others are more generic and can be notified using EiffelActivity\*Events.
+
+#### Examples of events related to activities:
+- [EiffelActivityTriggeredEvent](../eiffel-vocabulary/EiffelActivityTriggeredEvent.md)
+- [EiffelActivityStartedEvent](../eiffel-vocabulary/EiffelActivityStartedEvent.md)
+- [EiffelActivityFinishedEvent](../eiffel-vocabulary/EiffelActivityFinishedEvent.md)
+- [EiffelTestCaseTriggeredEvent](../eiffel-vocabulary/EiffelTestCaseTriggeredEvent.md)
+- [EiffelTestCaseStartedEvent](../eiffel-vocabulary/EiffelTestCaseStartedEvent.md)
+- [EiffelTestCaseFinishedEvent](../eiffel-vocabulary/EiffelTestCaseFinishedEvent.md)
+
+### Artifact
+_Artifacts_ are items or software packages generated in a CI/CD [pipeline](#pipeline), for example a built binary or a Docker image. An artifact should be possible to identify using a purl (package URL). An artifact is often the subject of a test executed or a delivery performed within a CI/CD [pipeline](#pipeline).
```

Comment:
This definition is enough, but maybe we could add that artifacts aren't always only things that will be included in products? 
They can also be test artifacts, i.e. artifacts only created to be useful in some specific test scenario. For example, when product software is combined with testing harnesses. Such artifacts are also defined by Baselines to record what versions of testing harness software were used.

Reply:
Agreed. Will add something along those lines.